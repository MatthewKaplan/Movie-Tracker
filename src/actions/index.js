export const fetchPopularMovies = popularMoviess => ({
  type: "FETCH_POPULAR_MOVIES",
  favorited: false,
  popularMoviess
});

export const fetchPopularTv = popularShows => ({
  type: "FETCH_POPULAR_TV",
  favorited: false,
  popularShows
});

export const fetchUpcoming = upcomingResults => ({
  type: "FETCH_UPCOMING",
  favorited: false,
  upcomingResults
});

export const fetchNews = newsResults => ({
  type: "FETCH_NEWS",
  newsResults
});

export const fetchGenres = genreResults => ({
  type: "FETCH_GENRES",
  favorited: false,
  genreResults
});

export const fetchSearch = searchResults => ({
  type: "FETCH_SEARCH",
  favorited: false,
  searchResults
});

export const setUser = user => ({
  type: "SET_USER",
  user
});

export const createUser = user => ({
  type: "CREATE_USER",
  user
});

export const isLoggedIn = bool => ({
  type: "IS_LOGGED_IN",
  isLoggedIn: bool
});

export const favoritesList = movies => ({
  type: "FAVORITES_LIST",
  favorited: false,
  movies
});

export const currentMovie = movie => ({
  type: "CURRENT_MOVIE",
  movie
});

export const handleError = errorMessage => ({
  type: "HANDLE_ERROR",
  errorMessage
});
