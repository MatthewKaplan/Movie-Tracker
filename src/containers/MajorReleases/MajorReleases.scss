@import "../../variables.scss";

.major-release {
  @include hover-state(8.5);
  background-size: cover;
  height: 525px;
  width: 350px;
  &:nth-child(2) {
    margin: 0 2.5rem 0 0.8rem;
  }
}