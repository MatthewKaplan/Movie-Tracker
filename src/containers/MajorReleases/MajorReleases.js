import React, { Component } from "react";
import "./MajorReleases.scss";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { currentMovie } from "../../actions/index";

export class MajorReleases extends Component {
  render() {
    const movieBackdrop = {
      backgroundImage: `url(https://image.tmdb.org/t/p/w500${this.props.img})`
    };

    return (
      <React.Fragment>
        <Link
          to="/MovieCardInfo"
          onClick={() => this.props.currentMovie(this.props.wholeObj)}
          className="major-release-section"
        >
          <div className="major-release" style={movieBackdrop} />
        </Link>
      </React.Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  currentMovie: movie => dispatch(currentMovie(movie))
});

export default connect(
  null,
  mapDispatchToProps
)(MajorReleases);
