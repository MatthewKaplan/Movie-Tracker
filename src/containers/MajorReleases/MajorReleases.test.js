import React from "react";
import { shallow } from "enzyme";
import { MajorReleases, mapDispatchToProps } from "./MajorReleases";
import * as actions from "../../actions/index";
import MockData from "../../assets/mockData";

const mockCurrentMovie = MockData.filteredMovies;
const currentMovieMock = jest.fn();
const mockWholeObj = MockData.filteredMovies;

describe("MajorReleases", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<MajorReleases wholeObj={mockWholeObj} img={""} currentMovie={currentMovieMock} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke 'currentMovie' with the correct param when clicked", () => {
    wrapper.find("[data-test='major-release']").simulate("click");
    expect(currentMovieMock).toHaveBeenCalledWith(mockWholeObj);
  })
});

describe("mapDispatchToProps", () => {
  it("should call dispatch for currentMovie", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = actions.currentMovie(mockCurrentMovie);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.currentMovie(mockCurrentMovie);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
