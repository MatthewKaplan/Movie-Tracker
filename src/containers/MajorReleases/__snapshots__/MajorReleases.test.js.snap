// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MajorReleases should match the snapshot 1`] = `ShallowWrapper {}`;
